common/src/stack/checklist/checklist.py

```diff
@@ -0,0 +1,466 @@
+
+				# Add current state to Redis
+				currState = backend.lastSuccessfulState()
+
+				if currState and not currState.isError and \
+					not currState.isAddedToRedis:
+					timeout = StateSequence.getTimeThreshold(currState.state, os, osver)
+					if timeout and self.r:
+						redisKey  = sm.ipAddr + '-' + currState.state.name
+						self.r.set(redisKey, currState.state.name, ex=timeout)
+						currState.isAddedToRedis = True
+
+			self.log.info('#### Installation Status Messages for %s ####' % sm.ipAddr)
+			for s in stateList:
+				self.log.info(s)
+			self.log.info('#############################################')
+
+			if sm.state == State.Reboot_Okay:
+				with self.lock:
+					stateList.clear()
+					backend.dhcpStateArr.clear()
+
+	def run(self):
+		self.shutdownFlag = threading.Event()
+
+		# Set logging parameters
+		self.log = logging.getLogger('checklist')
+		fh = logging.FileHandler('/var/log/checklist.log')
+		fileFormatter = logging.Formatter('%(message)s')
+		fh.setFormatter(fileFormatter)
+		fh.setLevel(logging.DEBUG)
+
+		if 'STACKDEBUG' in os.environ:
+			# STDOUT handler
+			consoleLogger = logging.StreamHandler(sys.stdout)
+			stdoutFormatter = logging.Formatter('%(message)s')
+			consoleLogger.setFormatter(stdoutFormatter)
+			consoleLogger.setLevel(logging.INFO)
+			self.log.addHandler(consoleLogger)
+
+		self.log.addHandler(fh)
+
+		if 'STACKDEBUG' in os.environ:
+			self.log.setLevel(logging.DEBUG)
+		else:
+			self.log.setLevel(logging.INFO)
+
+		self.log.info('Starting chklist...')
+		import redis
+		self.r = None
+		try:
+			self.r = redis.StrictRedis(host='localhost')
+		except:
+			self.log.error(self, 'cannot connect to redis')
+
+		self.ipBackendMap = {}
+		self.ignoreIpList = []
+		self.lock = threading.RLock()
+		self.getBackendInfo()
+
+		self.queue = queue.Queue()
+		context = zmq.Context()
+		tx = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+		self.mqProcessor = MQProcessor(context, tx, \
+			self.queue)
+		self.mqProcessor.setDaemon(True)
+		self.mqProcessor.start()
+
+		self.dhcpLog = LogParser(r'/var/log/messages', \
+			self.queue)
+		self.dhcpLog.setDaemon(True)
+		self.dhcpLog.start()
+
+		op = stack.api.Call('list.host.attr', ['a:frontend', 'attr=os'])
+		frontendOS = None
+
+		for o in op:
+			frontendOS = o['value']
+			break
+		self.ipThreadMap = {}
+		self.backendScript = None
+
+		if frontendOS == 'redhat':
+			self.run_redhat()
+		elif frontendOS == 'sles':
+			self.run_sles()
+
+		self.apacheLog.setDaemon(True)
+		self.apacheLog.start()
+
+		self.accessLog.setDaemon(True)
+		self.accessLog.start()
+
+		self.timeoutThread = CheckTimeouts(self.ipBackendMap, self.queue, self.lock)
+		self.timeoutThread.setDaemon(True)
+		self.timeoutThread.start()
+
+		self.processQueueMsgs()
+
+	def run_redhat(self):
+		self.apacheLog = LogParser(r'/var/log/httpd/ssl_access_log', \
+			self.queue)
+		self.accessLog = LogParser(r'/var/log/httpd/access_log', \
+			self.queue)
+		self.frontendOS = 'redhat'
```

Comment:
If you need to store the `frontendOS` in `self` then just do it in your `run` function, where you are pulling the data.